build/github/cockroach-microbench-ci.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+# Directories and file names
+output_dir="./artifacts/microbench"
+cleaned_current_dir="$output_dir/current"
+cleaned_base_dir="$output_dir/base"
+
+# hardcoded the file name as of now since `compare` requires a particular format for the file name.
+benchmark_file_name="pkg_sql-report.log"
+log_output_file_path="$output_dir/microbench.log"
+
+storage_bucket_url="gs://cockroach-microbench-ci"
+
+# Threshold for comparison
+threshold=$COMPARE_THRESHOLD
+exit_status=0
+
+# Configure Bazel and dev tooling
+cat <<EOF > ./.bazelrc.user
```

Comment:
I used this approach to build using `dev` as well in my script, but I've realised it's not the right approach for CI. Changes to `dev` could end up breaking things in the future. We should be using `run_bazel`. I have started updating my build script to this as well.
Here is an example from another script that builds `roachprod` & `roachtest` (`roachprod-microbench` can be built the same way): https://github.com/cockroachdb/cockroach/blob/master/build/teamcity/internal/release/process/roachtest-release-qualification.sh#L39
My pending changes: https://github.com/cockroachdb/cockroach/pull/123882/files#diff-95a0f7d2303cca3c0156c777c3a0044f180da20f77b6c6127178a1da10dd025aR32

Reply:
Github actions actually don't require `run_bazel` command for build commands.